### PR TITLE
Add more tests for bundle()

### PR DESCRIPTION
**Pull request by karenetheridge**
These are all the tests I was using for my implementation of bundle(), including a few interesting cases of name duplication, recursive references, etc.  All of my test cases are now implemented by `bundle` as of version 3.11, so hooray :)

### CODE REVIEW

File: t/more-bundle.t [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+use Test::More;
+use Test::Deep;
+use JSON::Validator;
+
+# these are triples:
+# - schema name to extract from schema file(s) with bundle()
+# - expected result
+# - test name.
+my @tests = (
```
Comment (jhthorsen) [inline t/more-bundle.t]: I'm pretty sure none of the files are run through perltidy.

File: t/more-bundle.t [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            type => 'object',
+            properties => {
+                gotcha => { '$ref' => '#/definitions/ref3' },
+            },
+        },
+        'find and resolve a reference that immediately leaps to another file',
+    ],
+);
+
+my $_validator = JSON::Validator->new;
```
Comment (jhthorsen) [inline t/more-bundle.t]: I find it very confusing following $jv, $_jv and $_validator. Can we name them something more descriptive?

File: t/more-bundle.t [hunk 3]
```diff
@@ -0,0 +1,264 @@
+use strict;
+use warnings;
```
Comment (jhthorsen) [inline t/more-bundle.t]: Please replace these two use statements with `use Mojo::Base -strict;`

File: t/more-bundle.t [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    cmp_deeply(
+        $_jv->schema->data,
+        $expected_output,
+        'our generated schema does not lose any data when parsed again by a validator',
+    );
+}
+foreach @tests;
+
+done_testing;
+# vim: set ts=4 sts=4 sw=4 et :
```
Comment (jhthorsen) [inline t/more-bundle.t]: I don't want $EDITOR config in the source files.

File: t/more-bundle.t [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    ],
+);
+
+my $_validator = JSON::Validator->new;
+$_validator->schema('http://json-schema.org/draft-07/schema#');
+
+my $jv = JSON::Validator->new;
+$jv->load_and_validate_schema('t/spec/more-bundle.yaml', { schema => 'http://json-schema.org/draft-07/schema#' });
+
+subtest $_->[2] => sub {
```
Comment (jhthorsen) [inline t/more-bundle.t]: I'm not a huge fan of subtests, since it makes it harder for me to figure out what has gone wrong when a smoker fail - at least I think that was my problem many years ago. Can you please change this to:

```
for my $t (@tests) {
  ...
}
```

File: t/more-bundle.t [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+    my $_jv = JSON::Validator->new;
+    $_jv->load_and_validate_schema($got, { schema => 'http://json-schema.org/draft-07/schema#' });
+    cmp_deeply(
+        $_jv->schema->data,
+        $expected_output,
+        'our generated schema does not lose any data when parsed again by a validator',
+    );
+}
+foreach @tests;
```
Comment (jhthorsen) [inline t/more-bundle.t]: Please use `for` instead of `foreach`.

**Comment by jhthorsen** [comment]
@karenetheridge: What's the status of this PR? I'll probably accept it and fix up the comments myself, if this PR is idling for much longer. (That means that I will be credited via "git blame" on many lines after running through perltidy)

**Comment by karenetheridge** [comment]
I plan to come back to this next week.

File: t/more-bundle.t (hunk 1 above)
Comment (karenetheridge) [reply t/more-bundle.t]: correct.

File: t/more-bundle.t (hunk 2 above)
Comment (karenetheridge) [reply t/more-bundle.t]: I'll rename the variables, to indicate what they're each used for.

File: t/more-bundle.t (hunk 5 above)
Comment (karenetheridge) [reply t/more-bundle.t]: The intent of the subtests is to make it *easier* to see what went wrong -- the tests are all the same for each block of data, so the subtest is labeled for which block of data is being tested.  Please take a look at the output again to see how much clearer it is with each dataset in its own section.

**Comment by karenetheridge** [comment]
I've force-pushed some changes.

File: t/more-bundle.t (hunk 1 above)
Comment (jhthorsen) [reply t/more-bundle.t]: Let me rephrase: All the code need to be run through "perltidy" to be accepted. You can use https://metacpan.org/pod/distribution/App-githook-perltidy/bin/githook-perltidy#githook-perltidy-install-[-force,-f] to do this automatically and locally for this repo.

File: t/more-bundle.t (hunk 5 above)
Comment (jhthorsen) [reply t/more-bundle.t]: Thanks for not giving up! The output is completely different from what I can remember, and I think I'll consider using subtests more places 👍

**Comment by jhthorsen** [comment]
Sorry, I missed out on the tidying. Thanks for the improved tests 👍
